src/main/groovy/org/mockito/release/notes/contributors/RecentContributorsFetcher.java

```diff
@@ -0,0 +1,118 @@
+        Set<Contributor> contributors = new LinkedHashSet<Contributor>();
+
+        try {
+            GitHubCommits commits = GitHubCommits
+                    .with(repository, readOnlyAuthToken, dateSince, dateUntil)
+                    .build();
+
+            while(commits.hasNextPage()) {
+                List<JsonObject> page = commits.nextPage();
+                contributors.addAll(extractContributors(page));
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Problems fetching commits from GitHub", e);
+        }
+
+        return contributors;
+    }
+
+    private Set<Contributor> extractContributors(List<JsonObject> commits) {
+        Set<Contributor> result = new HashSet<Contributor>();
+        for (JsonObject commit : commits) {
+            Contributor contributor = GitHubCommitsJSON.toContributor(commit);
+            if(contributor != null) {
+                result.add(contributor);
+            }
+        }
+        return result;
+    }
+
+    private static class GitHubCommits {
+
+        private final GitHubListFetcher fetcher;
+        private List<JsonObject> lastFetchedPage;
+
+        private GitHubCommits(String nextPageUrl) {
+            fetcher = new GitHubListFetcher(nextPageUrl);
+        }
+
+        boolean hasNextPage() {
+            return fetcher.hasNextPage();
+        }
+
+        List<JsonObject> nextPage() throws IOException, DeserializationException {
+            lastFetchedPage = fetcher.nextPage();
+            return lastFetchedPage;
+        }
+
+        static GitHubCommitsBuilder with(String repository, String readOnlyAuthToken, String dateSince, String dateUntil) {
+            return new GitHubCommitsBuilder(repository, readOnlyAuthToken, dateSince, dateUntil);
+        }
+
+        private static class GitHubCommitsBuilder {
+            private final String repository;
+            private final String readOnlyAuthToken;
+            private final String dateSince;
+            private final String dateUntil;
+
+            private GitHubCommitsBuilder(String repository, String readOnlyAuthToken, String dateSince, String dateUntil) {
+                notNull(repository, "repository", readOnlyAuthToken, "readOnlyAuthToken", dateSince, "dateSince");
+                this.repository = repository;
+                this.readOnlyAuthToken = readOnlyAuthToken;
+                this.dateSince = dateSince;
+                this.dateUntil = dateUntil;
+            }
+
+            GitHubCommits build() {
+                // see API doc: https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository
+                String nextPageUrl = "https://api.github.com/repos/" + repository + "/commits"
+                        + "?access_token=" + readOnlyAuthToken
+                        + "&since=" + dateSince
+                        + ((dateUntil != null)? "&until=" + dateUntil : "")
+                        + "&page=1&per_page=100";
```

Comment:
Considered, tried, but didn't seem more readable. Left it as is. I don't have strong opinion - if you wander in this part of the code and want to change it, go for it!